src/Mvc/Mvc.DataAnnotations/src/DataAnnotationsMetadataProvider.cs

```diff
@@ -30,6 +36,22 @@ internal class DataAnnotationsMetadataProvider :
         private const string NullableContextAttributeFullName = "System.Runtime.CompilerServices.NullableContextAttribute";
         private const string NullableContextFlagsFieldName = "Flag";
 
+        private static readonly HashSet<Assembly> _wellKnownFrameworkAssemblies = new HashSet<Assembly>
+        {
+            typeof(object).Assembly,
```

Comment:
Super nit: comment inline what assembly is in this set. Might help us figure out if we missed any here.

```suggestion
            typeof(object).Assembly, // System.Private.CoreLib
```